bciers/apps/reporting/src/app/components/changeReview/templates/facilityReportParser.ts

```diff
@@ -0,0 +1,325 @@
+import { ChangeItem } from "../constants/types";
+
+export function parseFacilityReportField(field: string): {
+  facilityName: string;
+  section: string;
+  activityName?: string;
+  sourceTypeName?: string;
+  unitIndex?: number;
+  fuelIndex?: number;
+  emissionIndex?: number;
+  fieldKey: string;
+  isFacilityLevel?: boolean;
+} | null {
+  // Updated regex to handle facility names with spaces and special characters
+  const facilityMatch = field.match(/\['facility_reports'\]\[(.*?)\]/);
+  if (!facilityMatch) return null;
+
+  // Remove any surrounding quotes from facility name
+  const facilityName = facilityMatch[1].replace(/(^['"])|(['"]$)/g, "");
+
+  const remainingPath = field.substring(
+    facilityMatch.index! + facilityMatch[0].length,
+  );
+
+  const pathSegments = remainingPath.match(/\[[^\[\]]*\]/g) || [];
+  const cleanSegments = pathSegments.map((segment) => {
+    const match = segment.match(/^\[(?:'|")?(.*?)(?:'|")?\]$/);
```

Comment:
would be helpful if we had a comment here explaining what the regex is looking for